### PR TITLE
Implement coin price

### PR DESCRIPTION
**Pull request by mduccc**

### CODE REVIEW

File: dig_core/lib/src/data/network/intercepter/dig_intecepter.dart [hunk 1]
```diff
@@ -23,12 +23,13 @@ class DigIntercepter extends InterceptorsWrapper {
   /// Trigger when [statusCode] = 200
   @override
   void onResponse(Response response, ResponseInterceptorHandler handler) {
-    /// Response data is not json, throw [BeanstalServerException]
-    if (response.data is! Map) {
+    /// Response data is not json or list, throw [BeanstalServerException]
```
Comment (manhduc-nguyen) [inline dig_core/lib/src/data/network/intercepter/dig_intecepter.dart]: what is `BeanstalServerException` ? pls change it if the name is not correct

File: dig_core/lib/src/domain/entity/staking/market.dart [hunk 2]
```diff
@@ -0,0 +1,160 @@
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'market.g.dart';
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class Market extends Equatable {
```
Comment (manhduc-nguyen) [inline dig_core/lib/src/domain/entity/staking/market.dart:7]: I often use this tool [quicktype](https://quicktype.io) for huge data classes. You know any tools that support better for dart ?

File: lib/app/cubit/staking/staking_cubit.dart [hunk 3]
```diff
@@ -30,10 +32,23 @@ class StakingCubit extends Cubit<StakingState> {
...
+    final Either<BaseDigException, ValidatorResponse>
+        getValidatorUseCaseResult =
+        results.first as Either<BaseDigException, ValidatorResponse>;
+    final Either<BaseDigException, List<Market>> _getCoinUseCaseResult =
+        results.last as Either<BaseDigException, List<Market>>;
+
+    num currentPrice = 0;
+    num priceChangePercentage24h = 0;
+
+    getValidatorUseCaseResult.fold((l) {
```
Comment (manhduc-nguyen) [inline lib/app/cubit/staking/staking_cubit.dart]: you should write it like `getValidatorUseCaseResult.fold((failure) {
      ...
    }, (markets) {
      ...
    });`

Its better reading than `l` `r`

File: dig_core/lib/src/data/network/intercepter/dig_intecepter.dart (hunk 1 above)
Comment (mduccc) [reply dig_core/lib/src/data/network/intercepter/dig_intecepter.dart:26]: I'm using only that tool:
https://javiercbk.github.io/json_to_dart/

File: dig_core/lib/src/domain/entity/staking/market.dart (hunk 2 above)
Comment (mduccc) [reply dig_core/lib/src/domain/entity/staking/market.dart:7]: > I often use this tool [quicktype](https://quicktype.io) for huge data classes. You know any tools that support better for dart ?

I'm using only that tool:
https://javiercbk.github.io/json_to_dart/

**Review by manhduc-nguyen — APPROVED** [review]
Great

**Pull request merged**